homeassistant/components/comfoconnect/sensor.py

```diff
@@ -72,264 +76,284 @@
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_MEASUREMENT,
+        name="Energy total",
+        unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        sensor_id=SENSOR_POWER_TOTAL,
+        last_reset=dt.utc_from_timestamp(0),
+    ),
+    ComfoconnectSensorEntityDescription(
+        key=ATTR_PREHEATER_POWER_CURRENT,
+        device_class=DEVICE_CLASS_POWER,
+        state_class=STATE_CLASS_MEASUREMENT,
+        name="Preheater power usage",
+        unit_of_measurement=POWER_WATT,
+        sensor_id=SENSOR_PREHEATER_POWER_CURRENT,
+    ),
+    ComfoconnectSensorEntityDescription(
+        key=ATTR_PREHEATER_POWER_TOTAL,
+        device_class=DEVICE_CLASS_ENERGY,
+        state_class=STATE_CLASS_MEASUREMENT,
+        name="Preheater energy total",
+        unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+        sensor_id=SENSOR_PREHEATER_POWER_TOTAL,
+        last_reset=dt.utc_from_timestamp(0),
+    ),
+)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_RESOURCES, default=[]): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
+            cv.ensure_list, [vol.In([desc.key for desc in SENSOR_TYPES])]
         )
     }
 )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the ComfoConnect fan platform."""
+    """Set up the ComfoConnect sensor platform."""
     ccb = hass.data[DOMAIN]
 
-    sensors = []
-    for resource in config[CONF_RESOURCES]:
-        sensors.append(
-            ComfoConnectSensor(
-                name=f"{ccb.name} {SENSOR_TYPES[resource][ATTR_LABEL]}",
-                ccb=ccb,
-                sensor_type=resource,
-            )
-        )
+    sensors = [
+        ComfoConnectSensor(ccb=ccb, description=description)
+        for description in SENSOR_TYPES
+        if description.key in config[CONF_RESOURCES]
+    ]
 
     add_entities(sensors, True)
 
 
 class ComfoConnectSensor(SensorEntity):
     """Representation of a ComfoConnect sensor."""
 
-    def __init__(self, name, ccb: ComfoConnectBridge, sensor_type) -> None:
+    _attr_should_poll = False
+    entity_description: ComfoconnectSensorEntityDescription
+
+    def __init__(
+        self,
+        ccb: ComfoConnectBridge,
+        description: ComfoconnectSensorEntityDescription,
+    ) -> None:
         """Initialize the ComfoConnect sensor."""
         self._ccb = ccb
-        self._sensor_type = sensor_type
-        self._sensor_id = SENSOR_TYPES[self._sensor_type][ATTR_ID]
-        self._name = name
+        self.entity_description = description
+        self._attr_name = f"{ccb.name} {self.entity_description.name}"
+        self._attr_unique_id = f"{self._ccb.unique_id}-{self.entity_description.key}"
 
     async def async_added_to_hass(self):
         """Register for sensor updates."""
         _LOGGER.debug(
-            "Registering for sensor %s (%d)", self._sensor_type, self._sensor_id
+            "Registering for sensor %s (%d)",
```

Comment:
Why is this needed?  `homeassistant.core` already have a debug log when entities are created

Reply:
It's not really needed, but it helps during debugging (that's why it's `_LOGGER.debug`). 
`homeassistant.core` is logging quite a lot, and that makes it hard to follow what sensor_id is registered.